### DISCUSSION TITLE
Receive wire:model initial value in component

### DISCUSSION THREAD

**Discussion by fontenele** [livewire/livewire#1109] [4. General]
"livewire/livewire": "1.1.0"

How can i get wire:model in component that have wire:model configured? 

Http/Livewire/UserForm.php
```php
class UserForm extends Component
{
    public $roles;
    public $user;

    public function mount(User $user)
    {
        $this->user = $user;
        $this->roles = $user->roles;
    }

    public function render()
    {
        return view('livewire.form.user-form');
    }
}
```

views/livewire/form/user-form.blade.php
```html
<div>
    <x-multiselect wire:model="roles"></x-multiselect>
</div>
```

views/components/multiselect.blade.php
```html
<div {{$attributes}} x-data="{roles: []}" x-ref="input" x-init="console.log($refs.input.value)">
...
</div>
```

I need to get in views/components/multiselect.blade.php the $roles value setted in mount() of UserForm.php, passed as wire:model to multiselect.

**Comment by fontenele** [comment]
Solved using setTimeout in x-init= to get $refs.input.value but this is not the good way...

**Comment by mblarsen** [comment]
You could pass something like `:initial-value` to your multiselect and then render the value from your blade component. Your Apline parts are to the component anyway.

```
@props(['initialValue' => [])
<div {{ $attributes }} x-data="{roles: @json($initialValue)}">
```

Didn't try this, but conceptually it should work.

**Reply by fontenele** [reply]
> You could pass something like `:initial-value` to your multiselect and then render the value from your blade component. Your Apline parts are to the component anyway.
> 
> ```
> @props(['initialValue' => [])
> <div {{ $attributes }} x-data="{roles: @json($initialValue)}">
> ```
> 
> Didn't try this, but conceptually it should work.

So, i have to use?

 <x-multiselect :initialValue="$roles" wire:model="roles"></x-multiselect>
